app/src/main/java/com/duckduckgo/app/browser/BrowserWebViewClient.kt

```diff
@@ -166,6 +168,57 @@ class BrowserWebViewClient @Inject constructor(
         super.onReceivedSslError(view, handler, error)
     }
 
+    @UiThread
+    override fun onReceivedHttpAuthRequest(view: WebView?, handler: HttpAuthHandler?, host: String?, realm: String?) {
+        Timber.v("onReceivedHttpAuthRequest ${view?.url} $realm, $host,  $currentUrl")
+
+        if (handler != null) {
+            Timber.v("onReceivedHttpAuthRequest - useHttpAuthUsernamePassword [${handler.useHttpAuthUsernamePassword()}]")
+            if (handler.useHttpAuthUsernamePassword()) {
+                val webViewDatabase = WebViewDatabase.getInstance(view?.context)
+
+                val credentials = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```

Comment:
might be neater to refactor this little block out to its own function

`buildAuthCredentials()` maybe? that would help hide the `SDK_INT` logic away and keep this function's logic clearer